src/service/telemetry-service.js

```diff
@@ -1,117 +1,97 @@
-}
-
-var dispatcher = undefined;
 
-if(localStorageEnabled === 'true') {
-    dispatcher = new Dispatcher(config);
-}
-
-const SUCCESS_RESP = {id: 'api.telemetry', ver: '1.0', ets: new Date().getTime(), params: {}, responseCode: "SUCCESS"};
-function ERROR_RESP(err) {
-    return {id: 'api.telemetry', ver: '1.0', ets: new Date().getTime(), params: {err: err}, responseCode: "SERVER_ERROR"};
-}
-
-exports.dispatch = function(req, res) {
-    
-    var message = req.body;
-    if(!message.mid) message.mid = uuidv1();
-    message.syncts = new Date().getTime();
-    const data = JSON.stringify(message);
-    const headers = {'authorization': 'Bearer ' + config.proxyAuthKey};
-    if (req.header('content-type'))
-        headers['content-type'] = req.header('content-type');
-    if (req.header('content-encoding'))
-        headers['content-encoding'] = req.header('content-encoding');
-    
-    if(localStorageEnabled === 'true' || telemetryProxyEnabled === 'true') {
-
-        if(localStorageEnabled === 'true') {
-            // Store locally and respond back with proper status code
-            dispatcher.dispatch(message.mid, data, function(err, data) {
-                if(err) {
-                    console.log('error', err);
-                    res.status(500)
-                    res.json(ERROR_RESP(err));
-                } else {
-                    res.json(SUCCESS_RESP);
-                }
-            });
-        }
-        if(localStorageEnabled === 'true' && telemetryProxyEnabled === 'true') {
-            // Store locally and proxy to the specified URL. If the proxy fails ignore the error as the local storage is successful. Do a sync later
-            var options = {url:config.proxyURL, headers: headers, body: data};
-            request.post(options, function optionalCallback(err, httpResponse, body) {
-                if (err) {
-                    return console.error('Proxy failed:', err);
-                }
-                console.log('Proxy successful!  Server responded with:', body);
-            });
-        } 
-        if(localStorageEnabled !== 'true' && telemetryProxyEnabled === 'true') {
-            // Just proxy
-            var options = {url:config.proxyURL, headers: headers, body: data};
-            request.post(options, function optionalCallback(err, httpResponse, body) {
-                if (err) {
-                    res.status(500)
-                    res.json(ERROR_RESP(err));
-                } else {
-                    console.log('Proxy successful!  Server responded with:', body);
-                    res.json(SUCCESS_RESP);
-                }
-            });
+class TelemetryService {
+    constructor(Dispatcher, config) {
+        this.config = config;
+        this.dispatcher = this.config.localStorageEnabled === 'true' ? new Dispatcher(config) : undefined;
+    }
+    dispatch(req, res) {
+        const message = req.body;
+        message.did = req.header('x-device-id');
```

Comment:
use [req.get](https://expressjs.com/en/4x/api.html#req.get) instead of req.header